lib/RestructureCFG/DAGifyPass.cpp

```diff
@@ -48,15 +49,15 @@ static void processRetreating(const revng::detail::EdgeDescriptor<BasicBlock *>
                               return Elem == Target;
                             }));
 
-  ScopeGraphBuilder SGBuilder(F);
   SGBuilder.makeGotoEdge(Source, Target);
 }
 
 class DAGifyPassImpl {
   Function &F;
+  ScopeGraphBuilder SGBuilder;
```

Comment:
I haven't checked, but it should be possible to make this a `const ScopeGraphBuilder`.

You probably have to change `ScopeGraphBuilder` itself to make all methods `const` but from a cursory look that's feasible too.

Let's strive for making `ScopeGraphBuilder` and its users more `const`-correct.

Reply:
OK, I made all the methods of `ScopeGraphBuilder` in this commit https://github.com/revng/revng/pull/480/commits/98f9098ddd7d7a2a9c4e2c6654e9e7696141d815.

Then, I made the field `const` in `DAGifyPass.cpp` here https://github.com/revng/revng/blob/3a9379ef740d1d2b9f7bcad02cc962719251e02a/lib/RestructureCFG/DAGifyPass.cpp#L25